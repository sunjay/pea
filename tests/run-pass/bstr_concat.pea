fn say_hello(name: [u8]) {
    println!(b"Hello, " + name + b".");
}

fn full_name(first: [u8], last: [u8]) -> [u8] {
    return first + b" " + last;
}

fn main() {
    let name1 = b"Manish";
    let name2 = b"Earth";
    let name3 = b"Moon";
    // Testing to make sure that string operations do not mutate the original string
    let full_name1 = full_name(name1, name2);
    let full_name2 = full_name(name1, name3);

    say_hello(full_name1);
    say_hello(full_name2);
}
