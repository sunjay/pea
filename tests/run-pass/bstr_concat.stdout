Hello, Manish Earth.
Hello, Manish Moon.
